### DISCUSSION TITLE
Server vs Client distinction

### DISCUSSION THREAD

**Discussion by aadamsx** [denoland/deno#9941] [Q&A] [answered]
How can we tell if we are on the server or in the browser?  In the Node world, we mostly just use the lack of a ```window``` object.  But of course Deno has this object.  What's a great, fool-proof way to tell in Deno?

I'm asking because I'm using Deno + SSR and would like to avoid the React useLayoutEffect warning like I'm getting when using Zustand: https://github.com/pmndrs/zustand/issues/347  in Aleph: https://github.com/alephjs/aleph.js/issues/221

**Comment by kitsonk** [comment, marked as answer]
```js
"Deno" in globalThis
```

**Comment by nuts-n-bits** [comment]
OK so today I learned that you can do react in deno. I'm impressed!! [Rocket emoji]
